pages/api/user/all.ts

```diff
@@ -37,32 +36,12 @@ export default async function handler(
   }
 }
 
-async function getUsers() {
+async function getAllUsers(args: { limit: number }) {
+  const { limit } = args;
   try {
-    const users: User[] = await prisma.user.findMany();
-
-    const usersWithProjects = await Promise.all(
-      users.map(async (user) => {
-        const projects = await prisma.project.findMany({
-          where: { authorId: user.id },
-        });
+    const users: User[] = await prisma.user.findMany({ take: limit });
 
-        const userProjects = projects.map((project) => project.title);
-
-        return {
-          id: user.id,
-          name: user.name,
-          title: user.title,
-          username: user.username,
-          skills: user.skills,
-          description: user.description,
-          profilePicture: user.image,
-          role: userProjects.length > 0 ? 'GS' : 'NS',
-          projects: userProjects,
-          numberOfProjects: userProjects.length,
-        };
-      })
-    );
+    const usersWithProjects = await getUsersWithProjects(users);
 
     const sortedUsers = usersWithProjects.sort(
```

Comment:
the sorting which was implemented earlier won't work now as we are fetching data in segments, need to sort it directly from db. if it's complex to implement, you can skip the sorting part. this can be done later with a seperate issue.

Reply:
sure, i will give it a try...